### PR TITLE
[0433] Handle visa collection in new course flow for scitts/heis

### PR DESCRIPTION
**Pull request by defong**
### Context
New course creation for Scitt and HEI 

### Changes proposed in this pull request

New course creation for scitts and hei 
along came the visa
during the visa creation

### Guidance to review
During the creation of course when apprenticeship is selected as `no` then there is student visa question
During the creation of course when apprenticeship is selected as `yes` then there is skilled worker visa question

A relative link is shown if the provider is HEI and do not sponsor for the visa in question
A relative link is shown if the provider is Scitt and do not sponsor for the skilled visa

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by defong** [comment]
[Handle visa collection in new course flow for scitts/heis](https://trello.com/c/vjBGt44v/433-handle-visa-collection-in-new-course-flow-for-scitts-heis)

**Comment by simonwhatley** [comment]
Student visas:

- don't show the 'Learn more about' link
- add the question: 'Can your organisation sponsor Student visas for this course?'

Screenshot 2022-09-14 at 11 14 17@2x

Check answers - Skilled Worker visas:

- 'Skilled Worder visa' should be plural 'Skilled Worker visas'

Screenshot 2022-09-14 at 11 17 30@2x

**Comment by simonwhatley** [comment]
When to show the 'Learn more about...' link:

| Provider type | Teaching apprenticeship | Visa sponsorship | Show learn more link? |
|---|---|---|---|
| HEI | Yes | No - Skilled Worker | Yes |
| HEI | No | No - Student | Yes |
| HEI | Yes | Yes - Skilled Worker | No |
| HEI | No | Yes - Student | No |
| SCITT | Yes | No - Skilled Worker | Yes |
| SCITT | No | No - Student | No |
| SCITT | Yes | Yes - Skilled Worker | No |
| SCITT | No | Yes - Student | No |

**Comment by simonwhatley** [comment]
In the add course flow, can you include 'Add course' as a caption above the page heading on every page?

File: app/views/publish/courses/skilled_worker_visa_sponsorship/_form_fields.html.erb [hunk 1]
```diff
@@ -6,11 +6,16 @@
...
     </legend>
 
-   <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">
-     Can your organisation sponsor Skilled Worker visas for this course?
-   </legend>
+    <% if @course.pg_teaching_apprenticeship? && !@provider.can_sponsor_skilled_worker_visa %>
+        <p class="govuk-body">Learn more about <%= govuk_link_to "recruiting trainee teachers from overseas", "https://www.gov.uk/guidance/recruit-trainee-teachers-from-overseas-accredited-itt-providers" %>.</p>
+    <% end %>
+
+    <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">
```
Comment (darokel) [inline app/views/publish/courses/skilled_worker_visa_sponsorship/_form_fields.html.erb:13]: I think the markup for this form is a bit dodgy. I thought I updated it but it looks i only fixed it for the student visas (if you rebase). For this one, i don't think the heading should be wrapped in a legend. Eg https://github.com/DFE-Digital/publish-teacher-training/blob/main/app/views/publish/courses/student_visa_sponsorship/_form_fields.html.erb#L2

File: app/views/publish/courses/skilled_worker_visa_sponsorship/_form_fields.html.erb [hunk 2]
```diff
@@ -6,11 +6,16 @@
       </h1>
     </legend>
```
Comment (darokel) [inline app/views/publish/courses/skilled_worker_visa_sponsorship/_form_fields.html.erb:7]: I think the markup for this form is a bit dodgy. I thought I updated it but it looks i only fixed it for the student visas (if you rebase). For this one, i don't think the heading should be wrapped in a legend. Eg https://github.com/DFE-Digital/publish-teacher-training/blob/main/app/views/publish/courses/student_visa_sponsorship/_form_fields.html.erb#L2

**Review by darokel — COMMENTED** [review]
Left just one comment. Think we should sort out the markup for the `app/views/publish/courses/skilled_worker_visa_sponsorship/_form_fields.html.erb`. Just need to move the heading out of the fieldset/legend so there aren't two

File: app/views/publish/courses/skilled_worker_visa_sponsorship/_form_fields.html.erb (hunk 2 above)
Comment (defong) [reply app/views/publish/courses/skilled_worker_visa_sponsorship/_form_fields.html.erb:7]: https://design-system.service.gov.uk/components/radios/default/index.html
You sure about h1 nested in fieldset?

Comment (defong) [reply app/views/publish/courses/skilled_worker_visa_sponsorship/_form_fields.html.erb]: moving h1 out

**Review by darokel — APPROVED** [review]

**Pull request merged**